### ISSUE TITLE
Would like to link to a different doc page

### ISSUE THREAD

**Issue by morethanreal** [PolymerElements/iron-doc-viewer#34]
I would like to refer to a the docs for a different element (potentially in the same repository) from a doc page.

**Comment by atotic** [comment]
Me too!

**Comment by mbleigh** [comment]
Is there a reason not to just use full markdown links?

**Comment by atotic** [comment]
How do we refer to docs for a different element? Will there be a canonical
url somewhere on polymer-project? Even then, there are multiple
versions.....

**Comment by mbleigh** [comment]
Yes, there will be canonical urls. I think versioning is something to worry
about down the road

**Comment by arthurevans** [comment]
See also #9. I think the correct solution from a _tool_ point of view is to allow the author to link to ArbitraryElement with some specific syntax, like [ArbitraryElement](component:ArbitraryElement) or
{@link ArbitraryElement} and have hydrolysis/a doc merger script/some other tool resolve those links when you generate a batch of docs.

**Comment by notwaldorf** [comment]
I think this already works with markdown, both with absolute and relative paths (to the current element). See https://github.com/PolymerElements/iron-icon/blob/master/iron-icon.html#L55 for an example.

**Comment by arthurevans** [comment]
This kinda works now, but it works more or less by accident. It works because in the catalog, each element has a path like `elements/foo-element`... Provided that the element is the main element in a package. (You can't, for example, link to `[paper-textarea](paper-textarea)` and have it work, because the URL for _that_ is `elements/paper-input?active=paper-textarea`.

You can embed the catalog-specific URL in your doc, but then you can't serve it statically, where the URL will be something like:

```
my-elements.html#iron-icons
```

And if you want to link to a specific property on a specific element, getouttatown.

So, my napkin-sketch of how this could work:

1. hydrolysis resolves {@link iron-input} or {@link iron-input#property-type} to a reference to a particular element or member and inserts (something) in the markup. 

Or:

1a. We insert the (something) directly in the markup, in the form of, say, a special URL:

```
  [iron-input](api:iron-input)
```

2. `iron-doc-viewer` intercepts anchor clicks for api: URLs and fires an event.

3. `iron-component-page` or the router of choice receives that event, translates it to the URL scheme in use, and navigates to it.

**Comment by arthurevans** [comment]
I'm fine with this being a low priority for now but I do think we need to solve it eventually. Probably blocked by: https://github.com/PolymerElements/iron-component-page/issues/57
